Talk about source ports (#392)
quicwg/ops-drafts: draft-ietf-quic-applicability.md

Decisions for those protocols might be informed by choices made by other
protocols, like HTTP/3.

## Source Port Selection

Some UDP protocols are vulnerable to reflection attacks, where an attacker is
able to direct traffic to a third party as a denial of service. For example,
these source ports are associated with applications known to be vulnerable to
reflection attacks (often due to server misconfiguration):

* port 53 - DNS {{?RFC1034}}
* port 123 - NTP {{?RFC5905}}
* port 1900 - SSDP {{SSDP}}
* port 5353 - mDNS {{?RFC6762}}
* port 11211 - memcached

Services might block source ports of protocols known to be vulnerable to
reflection, to avoid the overhead of processing large numbers of packets by
their QUIC implementations. However, this practice has negative effects on
clients: not only does it require establishment of a new connection, but in
some instances, might cause the client to avoid using QUIC for that service for

> a period of time, downgrading to a non-UDP protocol.

Reviewer: ```suggestion
a period of time, downgrading to a non-UDP protocol (see {{fallback}}).
```

Decisions for those protocols might be informed by choices made by other
protocols, like HTTP/3.

## Source Port Selection

Some UDP protocols are vulnerable to reflection attacks, where an attacker is
able to direct traffic to a third party as a denial of service. For example,
these source ports are associated with applications known to be vulnerable to
reflection attacks (often due to server misconfiguration):

* port 53 - DNS {{?RFC1034}}
* port 123 - NTP {{?RFC5905}}
* port 1900 - SSDP {{SSDP}}
* port 5353 - mDNS {{?RFC6762}}
* port 11211 - memcached

Services might block source ports of protocols known to be vulnerable to
reflection, to avoid the overhead of processing large numbers of packets by

> their QUIC implementations. However, this practice has negative effects on

Author: I don't think it's the port that's vulnerable to attacks; it's the underlying protocol.

> Services might block source ports of protocols known to be vulnerable to
> reflection, to avoid the overhead of processing large numbers of packets by
> their QUIC implementations. However, this practice has negative effects on

Reviewer 2: ```suggestion
Services might block source ports associated with protocols known to be
vulnerable to reflection attacks, to avoid the overhead of processing large
numbers of packets. However, this practice has negative effects on
```

> their QUIC implementations. However, this practice has negative effects on

Reviewer 2: split the difference between Ian's and Mark's suggestions?

clients: not only does it require establishment of a new connection, but in
some instances, might cause the client to avoid using QUIC for that service for
a period of time, downgrading to a non-UDP protocol (see {{fallback}}).

As a result, client implementations are encouraged to avoid using source ports
associated with protocols known to be vulnerable to reflection attacks. Note

> that the list above is only indicative; other source ports might be considered

Reviewer 3: ```suggestion
that the list above is not exhaustive; other source ports might be considered
```
